### ISSUE TITLE
Embedding Pynecone in other apps

### ISSUE THREAD

**Issue by EgosOwn** [pynecone-io/pynecone#46]
One of the issues i had with streamlit (from what i remember) is that you can't easily embed streamlit into existing applications, and the same seems to be true here. 

It would be really cool if i could do pynecone.run() from an existing app and bind state to existing modules. It would one to take existing APIs and wire them up quickly to a pyonecone GUI.

I know that is certainly easier said than done. Go ahead and close this issue if that is too far out of scope or isn't feasible.

Thank you for your efforts, pynecone is neat.

**Comment by picklelo** [comment]
This isn't on our roadmap currently, but is a really interesting use case. We will think about how we can incorporate this in the future. Thanks for checking us out!

**Comment by Alek99** [comment]
Yeah really cool suggestion there may be some ways to do this, we'll brainstorm

**Comment by chapman73** [comment]
Since pynecone uses FastAPI, it would be useful if it could support patterns via [sub applications](https://fastapi.tiangolo.com/advanced/sub-applications/).

Something like:

```
pc_app = pc.App(state=State)

app = FastAPI()
app.mount("/pynecone", pc_app)
```

**Comment by Alek99** [comment]
Ah I see, this is an interesting idea.  I'm not too familiar with sub application feature but Ill take get back about this! Thanks for the suggestion

**Comment by kkinder** [comment]
Just dropping a thought here. There's no admin interface on FastAPI, so Pynecone could have a useful application in developing what amount to admin/support interfaces. For example, if you built an API, but you want something like Django-style admin abilities, you could build that pretty easily (I imagine) in Pynecone if it were a supported use case.

**Comment by jerguslejko** [comment]
my 2c: I have couple of 'utility apps' running on out k8s cluster at work - these apps are mainly for internal use (one form for QA to run periodic tasks manually on testing environment, or some UIs to help with on-call duty). They are usually "single form apps", just do one thing. the stack we use for this is fastapi, installed as a helm chart. the only thing that is "hard" or "icky" to do in these apps is the UI.

so being able to do `app.mount("/pynecone", pc_app)` would be a game changer, really!

**Comment by picklelo** [comment]
That makes a lot of sense, it would be great to combine all these "microapps" together into a single service rather than hosting each one separately. We're seeing a similar use case with our website as we're trying to add a blog subapp to it, but we want to host it along with our main docs site.

We will definitely look into this soon!

**Comment by adhami3310** [comment]
hey everyone! i know this is an old ticket, but we have changed a lot since then (for one, we are now reflex).

it's possible to put reflex alongside other fastapi apps or similar: https://reflex.dev/docs/api-routes/overview/

if you want to not bundle the frontend separately, you can take a look into reflex-enterprise (free, but has a badge for free users of reflex cloud): https://reflex.dev/docs/enterprise/overview#reflex-enterprise

what i read from the above conversation does mean that this one is completed, but i will leave it open a bit if the above is not sufficient

**Comment by masenf** [comment]
Leaving this open because it seems like it would work in theory, but in practice it's complicated and tricky to actually mount a reflex app as a sub-asgi:

* app gets compiled to js, but no asgi-only trigger to actually make a production static files build, so there is no frontend to serve. this is less of an issue for larger deployments where the backends would run separately.
* `__REFLEX_MOUNT_FRONTEND_COMPILED_APP=true` has weird interplay with `REFLEX_FRONTEND_URL` where the CSS and assets can load through the mounted path, but the app is requesting them from the root.
* lifespan tasks may be lost when mounting a reflex app this way

i'm not sure where the indirection needs to happen, but it should be super simple to just mount a reflex app on a sub path, and even now in 2026, it's just not simple.
